### ISSUE TITLE
Cannot go to method with cmd-r

### ISSUE THREAD

**Issue by larryzhao** [rubyide/vscode-ruby#321]
### Your environment

- `vscode-ruby` version: 0.17.0
- Ruby version: 2.4.2
- VS Code version: Version 1.22.1 (1.22.1)
- Operating System: macOS 10.13.3
- Hardware (optional): 

Make sure you have `ruby`, `ruby-debug-ide` and `ruby-debug-basex19` installed before submitting your issue -- thank you !

### Expected behavior
I could navigate to methods in ruby file with cmd-r

### Actual behavior
`cmd-r` reports `No symbol information for the file`

### Steps to reproduce the problem
I think it's doing wrong after the recent VSCode upgrade.

**Comment by douglascamata** [comment]
I'm also running the latest vscode release since the beginning of my work day and I cannot reproduce.

**Comment by larryzhao** [comment]
@douglascamata I've also tried to uninstall `vscode-ruby` and reinstall it again. Still not working? Is there anything else I could try? Reinstall VSCode?

**Comment by douglascamata** [comment]
Are you using the `rubyLocate` or `solargraph` method for the go to method?

**Comment by larryzhao** [comment]
@douglascamata No. Just use `cmd - r` to navigate to methods defined in the current file is not working now.

**Comment by douglascamata** [comment]
Try checking manually in the terminal if you have the `solargraph` binary.

**Comment by larryzhao** [comment]
@douglascamata Sorry, I have `solargraph`. It seems it comes in as the dependency of Extension Ruby. I didnt' know it.

**Comment by douglascamata** [comment]
I guess you will need to check for errors in the developer's tools console then. :/

**Comment by gurgeous** [comment]
Hi guys. Yes, solargraph is a dependency in the current release. We like solargraph and generally recommend trying it. The next release will remove the dependency and just supply a suggestion instead.

**Comment by larryzhao** [comment]
@douglascamata 

Thx, I could see `[Extension Host] The Solargraph gem is not available.` in the developer tools.
I install `Solargraph` under `ruby-2.4.2@global` gemset, and change the setting to point to it: `"solargraph.commandPath": "/Users/larry/.rvm/rubies/ruby-2.4.2/lib/ruby/gems/2.4.0/bin/solargraph"` and reload vscode, but still go no symbols.

**Comment by douglascamata** [comment]
Did solargraph finish analysing your code?

**Comment by larryzhao** [comment]
@douglascamata I don't think it matters, because when I open developer tools, it still says `[Extension Host] The Solargraph gem is not available.`

**Comment by douglascamata** [comment]
There must be something wrong in your environment variables, imho.

**Comment by tejo** [comment]
same here, after update. 
Same env and sw versions as @larryzhao except ruby 2.5.1

**Comment by wingrunr21** [comment]
@castwide any thoughts here?

**Comment by gurgeous** [comment]
We really need to add a section to the FAQ about gem loading... we will have install issues for every gem dependency.

One way to debug these problems is investigating/installing solargraph from within VS Code's Integrated Terminal. (View > Integrated Terminal). Try `ruby -v`, `gem env gemdir`, `gem list | grep solargraph`, `which solargraph` and then `solargraph -v`. This might shed some light.

**Comment by castwide** [comment]
First, I'd try running the commands in the VS Code terminal as @gurgeous suggested. If `solargraph` works from there, you should be able to run it without setting the `solargraph.commandPath` configuration.

Also, make sure the `ruby.intellisense` configuration isn't set to `none`.

**Comment by wingrunr21** [comment]
PRs welcome for docs. The Wiki could probably use some updates too

**Comment by gurgeous** [comment]
See https://github.com/rubyide/vscode-ruby/pull/327. I know this issue is specifically around `solargraph`, but a lot of folks have these problems with rubocop too. I wonder if we should revisit the config. Right now we have `useBundler`, `pathToRuby`, `pathToBundler`, `pathToRDebugIDE`, etc. Might be easier to just let users specify the full command line for each tool:

```
$HOME/.rbenv/shims/ruby bundle exec rubocop
```

This is how I setup the config for my other extension. Just a thought.

**Comment by castwide** [comment]
I just published version 0.15.0 of vscode-solargraph, which completes the separation of concerns with vscode-ruby; i.e., one's configuration no longer depends on the other. This might help resolve some of the problems and confusion. For people who aren't using Solargraph, it's also now possible to disable or uninstall the extension.

**Comment by Hbbb** [comment]
## Environment
vscode-ruby version: 0.18.0
Ruby version: 2.5.1
VS Code version: Version 1.25.1
Operating System: macOS 10.12.6

FYI, I am seeing this issue on only some files. Here is a screenshot of the devtools console when I hit `cmd+r` on a file where symbol definitions fail:

<img width="1627" alt="developer_tools_-_file____applications_visual_20studio_20code_app_contents_resources_app_out_vs_workbench_electron-browser_bootstrap_index_html_config__7b_22__22_3a_5b_22_2fusers_2fharrisonborges_2fprojects_2fgithub_com_2ffreshly_2ffreshly_" src="https://user-images.githubusercontent.com/3803272/42959990-99c9fb8c-8b58-11e8-9802-9b9f6bc37612.png">
